### PR TITLE
qa/tasks/tox: use python3 for tox tests

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- changelog
  - use python3 for testing
  - use newer tox version
  - use newer tempest and keystone versions
  - use `keystone-manage bootstrap` for prefilling
  - use V3 identity APIs
  - use username/password based authentication instead of admin token
  - unpin python-openstackclient
  - update the blacklist of tempest tests
- Fixes https://tracker.ceph.com/issues/45691
- tested at
  - http://pulpito.ceph.com/kchai-2020-05-30_17:46:31-rgw-wip-kefu-testing-2020-05-27-2242-distro-basic-smithi/5105719/
  - http://pulpito.ceph.com/kchai-2020-06-02_01:49:31-rgw-master-distro-basic-smithi/

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
the backport of this change should partially fix http://pulpito.ceph.com/kchai-2020-05-24_17:10:51-rgw-nautilus-distro-basic-smithi/

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2020-05-28_04:39:12-rgw-wip-kefu-testing-2020-05-27-2242-distro-basic-smithi/

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2020-05-28_06:03:00-rgw-wip-kefu-testing-2020-05-27-2242-distro-basic-smithi/

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2020-05-28_06:44:58-rgw-wip-kefu-testing-2020-05-27-2242-distro-basic-smithi/

**Comment by tchaikov** [comment]
@rzarzynski hi Radek, could you help re-review this change? i managed to fix the keystone based tests. because it's blocking #34171, and current tests using tempest against master is broken, see http://pulpito.ceph.com/cbodley-2020-05-29_16:33:44-rgw-master-distro-basic-smithi

it'd be great if we can get it merged sooner.

File: qa/tasks/keystone.py [hunk 1]
```diff
@@ -299,39 +306,41 @@ def run_section_cmds(ctx, cclient, section_cmd, specials,
             [ '--debug' ])
 
 def create_endpoint(ctx, cclient, service, url, adminurl=None):
-    endpoint_section = {
-        'service': service,
-        'publicurl': url,
-    }
+    endpoint_sections = [
+        {'service': service, 'interface': 'public', 'url': url},
```
Comment (rzarzynski) [inline qa/tasks/keystone.py:307]: ACK.

File: qa/tasks/keystone.py [hunk 2]
```diff
@@ -299,39 +306,41 @@ def run_section_cmds(ctx, cclient, section_cmd, specials,
...
+                     endpoint_sections)
 
 @contextlib.contextmanager
 def fill_keystone(ctx, config):
     assert isinstance(config, dict)
 
     for (cclient, cconfig) in config.items():
         # configure tenants/projects
+        run_section_cmds(ctx, cclient, 'domain create', 'name',
+                         cconfig['domains'])
```
Comment (rzarzynski) [inline qa/tasks/keystone.py]: ACK.

File: qa/tasks/keystone.py [hunk 3]
```diff
@@ -260,25 +263,25 @@ def run_keystone(ctx, config):
...
-def dict_to_args(special, items):
+def dict_to_args(specials, items):
     """
     Transform
         [(key1, val1), (special, val_special), (key3, val3) ]
     into:
         [ '--key1', 'val1', '--key3', 'val3', 'val_special' ]
     """
-    args=[]
+    args = []
```
Comment (rzarzynski) [inline qa/tasks/keystone.py:273]: Thanks fixing this pep8 abuse.

File: qa/tasks/tempest.py [hunk 4]
```diff
@@ -60,15 +60,6 @@ def download(ctx, config):
         sha1 = cconf.get('sha1')
         if sha1 is not None:
             run_in_tempest_dir(ctx, client, [ 'git', 'reset', '--hard', sha1 ])
-
-        # tox.ini contains a dead link, replace it with the new one
```
Comment (rzarzynski) [inline qa/tasks/tempest.py:64]: ACK.

File: qa/tasks/keystone.py [hunk 5]
```diff
@@ -322,27 +323,40 @@ def fill_keystone(ctx, config):
...
+                'project-name': 'admin',
+                'role-name': 'admin',
+                'service-name': 'keystone',
+                'region-id': 'RegionOne',
+                'admin-url': admin_url,
+                'public-url': url}
+        bootstrap_args = chain.from_iterable(('--bootstrap-{}'.format(k), v)
+                                             for k, v in opts.items())
+        run_in_keystone_venv(ctx, cclient,
+                             ['keystone-manage', 'bootstrap'] +
```
Comment (rzarzynski) [inline qa/tasks/keystone.py:339]: ACK.

File: qa/suites/rgw/tempest/tasks/rgw_tempest.yaml [hunk 6]
```diff
@@ -38,17 +38,16 @@ tasks:
         reseller_admin_role: admin
       object-storage-feature-enabled:
         container_sync: false
-        discoverability: false
+        discoverability: true
```
Comment (rzarzynski) [inline qa/suites/rgw/tempest/tasks/rgw_tempest.yaml:42]: ACK.

File: qa/tasks/tempest.py [hunk 7]
```diff
@@ -181,13 +181,17 @@ def task(ctx, config):
         ceph:
           conf:
             client:
-              rgw keystone admin token: ADMIN
```
Comment (rzarzynski) [inline qa/tasks/tempest.py:193]: ACK.

File: qa/tasks/keystone.py [hunk 8]
```diff
@@ -140,9 +140,7 @@ def setup_venv(ctx, config):
             ])
 
         run_in_keystone_venv(ctx, client,
-            [   'pip', 'install', 'python-openstackclient<=3.19.0',
-                '-r', 'requirements.txt'
-            ])
+            [   'pip', 'install', 'python-openstackclient'])
```
Comment (rzarzynski) [inline qa/tasks/keystone.py:143]: OK.

File: qa/tasks/tempest.py [hunk 9]
```diff
@@ -159,10 +159,8 @@ def run_tempest(ctx, config):
                 get_tempest_dir(ctx) + '/workspace.yaml',
                 '--workspace',
                 'rgw',
-                '--regex',
-                    '(tempest.api.object_storage)' +
-                    ''.join([ '(?!{blackitem})'.format(blackitem=blackitem)
-                        for blackitem in blacklist])
+                '--regex', '^tempest.api.object_storage',
+                '--black-regex', '|'.join(blacklist)
```
Comment (rzarzynski) [inline qa/tasks/tempest.py:163]: :+1:

**Pull request merged**